src/awk.h

```diff
@@ -22,11 +22,16 @@
  */
 #ifndef AWK_H
 #define AWK_H
-#ifdef HAVE_PCRE_H
-  #include <pcre.h>
-#endif
-#ifdef HAVE_PCRE_PCRE_H
-  #include <pcre/pcre.h>
+#ifdef XASTIR_LEGACY_PCRE
+  #ifdef HAVE_PCRE_H
+    #include <pcre.h>
+  #endif
+  #ifdef HAVE_PCRE_PCRE_H
+    #include <pcre/pcre.h>
+  #endif
+#else
+  #define PCRE2_CODE_UNIT_WIDTH 8
```

Comment:
Is this needed both here and in the autoconf rule?

Reply:
Yes, it is needed in both places.

"pcre2.h" throws an error if you include it without defining PCRE2_CODE_UNIT_WIDTH.

The AC_CHECK_HEADERS macro tries to compile small test program that includes the header you request.  The macro is:
   AC_CHECK_HEADERS([header to check],[action if found],[action if not found],[extra stuff to define/include first])

The "#define PCRE2_CODE_UNIT_WIDTH 8" thing in that call *only* sticks that define in the little test program.

So if we need to include pcre2.h anywhere else, you MUST also define it before you include.